doc/declarative-state-allocation/rds.md

```diff
@@ -0,0 +1,146 @@
+Allocate RDS instances using OAM definition
+===============
+
+This document covers the capability to declaratively allocate state using AWS RDS (Relational Database Service) instance, this capability uses [CrossPlane](https://crossplane.io/) addon to create cloud resources using straightforward Application definitions from OAM/kubeVela.
+
+
+Creating a new instance
+---------------
+
+To create a new instance we have to add a ***rds-instance*** trait with allocateDatabase property as follows:
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: rds-test-owner
+  namespace: default
+spec:
+  components:
+  - name: rds-test-owner
+    properties:
+      cpu: "1"
+      exposeType: ClusterIP
+      image: nginx
+      memory: 1024Mi
+      ports:
+      - expose: true
+        port: 80
+        protocol: TCP
+    traits:
+    - type: scaler
+      properties:
+        replicas: 1
+    - type: rds-instance
+      properties:
+        # This trait definition will create a new RDS instance
+        allocateDatabase:
+          storage: 400
+          instanceClass: db.t3.medium
```

Comment:
We should link to instance types, and perhaps give some recommendations for various workload types.